git_machete/client.py

```diff
@@ -2327,3 +2333,16 @@ def __sync_before_creating_pr(self, *, opt_onto: Optional[LocalBranchShortName],
             if ans in ('y', 'yes'):
                 return
             raise MacheteException('Pull request creation interrupted.')
+
+    def delete_untracked(self, opt_yes: bool) -> None:
+        branches_to_delete: List[LocalBranchShortName] = []
+        for managed_branch in self.managed_branches.copy():
+            status, _ = self.__git.get_strict_remote_sync_status(managed_branch)
+            if status == SyncToRemoteStatuses.UNTRACKED:
+                if managed_branch not in self.__down_branches:
```

Comment:
Hmmm it's theoretically possible that `self.__down_branches[managed_branch]` is defined but empty 🤔 I'd need to think hard if such a scenario can _actually_ happen in the current implementation... but I won't rely on such assumptions anyway

Reply:
(ofc this would only be a **false negative**: branch NOT slid out even though it should, not a false positive) which would be worse

Reply:
@PawelLipski

Reply:
In a nutshell, how about:
```suggestion
                if not self.__down_branches.get(managed_branch):
```

Reply:
This would also cover the case when `self.__down_branches[managed_branch]` is `None` or `[]` 🤔